phaser/pipeline.py

```diff
@@ -144,6 +153,41 @@ def __init__(self, working_dir=None, source=None, phases=None):
         self.phases = phases or self.__class__.phases
         self.phase_instances = []
         self.context = Context(working_dir=self.working_dir)
+        # Collect the extra sources and outputs from the phases so they can be
+        # reconciled and initialized as necessary.  Extra sources that match
+        # with extra outputs do not need to be initialized, but extra sources
+        # that do not come from phases must be initialized so that phases can
+        # access their data.
+        self.extra_sources = [
+            source for phase in self.phases for source in Pipeline._find_extra_sources(phase)
+        ]
+        self.extra_outputs = [
+            output for phase in self.phases for output in Pipeline._find_extra_outputs(phase)
+        ]
+        self.sources_needing_initialization = [
+            source
+            for source in self.extra_sources
+            if source not in self.extra_outputs
+        ]
+
+    def _find_extra_sources(phase):
+        if hasattr(phase, 'extra_sources'):
+            return phase.extra_sources
+        return []
+
+    def _find_extra_outputs(phase):
+        if hasattr(phase, 'extra_outputs'):
+            return phase.extra_outputs
+        return []
+
+    def init_source(self, name, source):
+        """ Initializes a named source based on the kind of 'source' passed in.
+
+        :param source: must be a os.PathLike file in csv format and will be read entirely into memory
+        """
+        # TODO: Check that file exists first?
+        data = read_csv(source).to_dict('records')
```

Comment:
As we gather extra methods that read data, we should allow the Pipeline to have a method for loading data that is not called "read_csv" and can be overridden when somebody customizes the Pipeline.

That can happen later, of course, and even later than that, we will probably have some use cases where 'employees.csv' is loaded as a CSV but 'pay_api_data' is loaded as JSON... then merged...